### PR TITLE
[Ubuntu] Set user-related environmental variables and PATH via post deployment script

### PR DESCRIPTION
**Pull request by miketimofeev**
# Description
It turned out that we can't use the `$BASH_ENV` variable because the values from it can't be overridden by workflow's `env` block.
This PR introduces another approach in which we set the appropriate variables during the machine provisioning process via a post-generation script.

#### Related issue:
https://github.com/actions/virtual-environments/issues/2954

## Check list
- [ ] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/linux/post-generation/environment-variables.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+# Replace $HOME with the default user's home directory for environmental variables related to the default user home directory
+
+default_user=$(cut -d: -f1 /etc/passwd | tail -1)
```
Comment (al-cheb) [inline images/linux/post-generation/environment-variables.sh:5]: We can request user home folder:
`homeDir=$(cut -d: -f6 /etc/passwd | tail -1)`

Comment (miketimofeev) [reply images/linux/post-generation/environment-variables.sh]: Thanks!

**Review by maxim-lobanov — APPROVED** [review]

**Review by maxim-lobanov — APPROVED** [review]

**Comment by EmFl** [comment]
What about changes from #2826 ? 
Right now on my system I get:
/home/ubuntu/$HOME/.config/configstore
/etc/environment contains XDG_CONFIG_HOME=$HOME/.config

Was the rollback intentional ?

**Comment by miketimofeev** [comment]
@EmFl yes it was because $BASH_ENV causes some issues since updated PATH is not propagated to other shells like pwsh. 
Please consider running post-deployment scripts on your system — this one substitutes the values in /etc/environment with the actual username https://github.com/actions/virtual-environments/blob/main/images/linux/post-generation/environment-variables.sh

**Pull request merged**